src/Security/Authentication/Core/src/AuthenticationHandler.cs

```diff
@@ -131,6 +154,11 @@ public async Task InitializeAsync(AuthenticationScheme scheme, HttpContext conte
 
         Options = OptionsMonitor.Get(Scheme.Name);
 
+        TimeProvider = Options.TimeProvider ?? TimeProvider.System;
+#pragma warning disable CS0618 // Type or member is obsolete
+        Clock ??= new SystemClock(TimeProvider);
```

Comment:
Will Clock ever actually be null here given DI is going to use the largest satisfiable constructor and and `AddAuthentication` still adds an `ISystemClock` to DI?

```suggestion
        Clock = new SystemClock(TimeProvider);
```